dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/openaire/OpenAireExportUtil.java

```diff
@@ -768,73 +724,55 @@ private static void writeGrantElement(XMLStreamWriter xmlw, GrantInfo grantInfo,
-
-            if (givenName != null && !isOrganization) {
-                contributor_map.put("nameType", "Personal");
-                writeFullElement(xmlw, null, "contributorName", contributor_map, contributorName, language);
-
-                String familyName = "";
-                if (givenName.length() + 1 < contributorName.length()) {
-                    familyName = contributorName.substring(givenName.length() + 1);
-                }
-
-                writeFullElement(xmlw, null, "givenName", null, givenName, language);
-                writeFullElement(xmlw, null, "familyName", null, familyName, language);
-            } else {
-                // default
-                if (isOrganization || ("ContactPerson".equals(contributorType) && !isValidEmailAddress(contributorName))) {
-                    contributor_map.put("nameType", "Organizational");
-                }
-                writeFullElement(xmlw, null, "contributorName", contributor_map, contributorName, language);
-            }
+        if (givenName != null && !isOrganization) {
+            contributor_map.put("nameType", "Personal");
+        } else if ((isOrganization || "ContactPerson".equals(contributorType)) && !isValidEmailAddress(contributorName)) {
+            contributor_map.put("nameType", "Organizational");
         }
+        writeFullElement(xmlw, null, "contributorName", contributor_map, contributorName, language);
+
+        writeGivenNameAndFamilyNameElements(xmlw, contributorName, language, givenName, isOrganization);
 
         if (StringUtils.isNotBlank(contributorAffiliation)) {
             writeFullElement(xmlw, null, "affiliation", null, contributorAffiliation, language);
         }
         xmlw.writeEndElement(); // </contributor>
     }
 
+    private static void writeGivenNameAndFamilyNameElements(XMLStreamWriter xmlw, String contributorName, String language,
+                                                            String givenName, boolean isOrganization) throws XMLStreamException {
+        if (contributorName.contains(",") && !isOrganization && (!contributorName.replaceFirst(",", "").contains(","))) {
+            String[] fullName = contributorName.split(", ");
+            givenName = fullName[1].trim();
+            String familyName = fullName[0].trim();
+
+            writeFullElement(xmlw, null, "givenName", null, givenName, language);
+            writeFullElement(xmlw, null, "familyName", null, familyName, language);
+        } else if (givenName != null && !isOrganization) {
+            String familyName = "";
+            if (givenName.length() + 1 < contributorName.length()) {
+                familyName = contributorName.substring(givenName.length() + 1);
+            }
+
+            writeFullElement(xmlw, null, "givenName", null, givenName, language);
+            writeFullElement(xmlw, null, "familyName", null, familyName, language);
```

Comment:
I think that we should incorporate extraction of family name in the code that extracts first name. I would recommend to change FirstNames.getFirstName to FirstNames.extractFirstAndLastName. It could look like something like this:
```
    public Tuple2<String, String> extractFirstAndLastName(String fullname) {
        int commaCount = StringUtils.countMatches(fullname, ",");
        if (commaCount == 0) {
            String[] words = fullname.split("\\s+");
            int firstNameWordsCount = 0;

            for (int i=0; i<words.length; ++i) {
                String word = words[i];
                boolean isLastWord = (i == (words.length - 1));
                boolean isFirstWord = (i == 0);

                if (isWordPartOfFirstName(word, isFirstWord, isLastWord)) {
                    firstNameWordsCount = i+1;
                } else {
                    break;
                }

            }
            
            return new Tuple2<String, String>(
                    StringUtils.join(words, ' ', 0, firstNameWordsCount),
                    StringUtils.join(words, ' ', firstNameWordsCount, words.length));

        } else if (commaCount == 1 && !fullname.endsWith(",")) {
            String[] words = fullname.split(",");
            return new Tuple2<>(words[1].trim(), words[0].trim());
        }
        return null;
    }

    private boolean isWordPartOfFirstName(String word, boolean isFirstWord, boolean isLastWord) {
        boolean wordIsAFirstName = isFirstName(word);
        
        if (wordIsAFirstName) {
            if (isFirstWord && isLastWord) {
                return true;
            }
            if (!isLastWord && !isStartOfLastName(word)) {
               return true;
            }
        }
        return false;
    }
```
It would work very similar to the original one mixed with the logic that is written in this method.
The difference is that it will not split fullname in many places, and it would have to be executed only for strings that we know that it is a fullname.

Reply:
It keeps every test case from the tests that it was written before:
```
  @ParameterizedTest
    @CsvSource(delimiter = '|', value = {
            "Awesome, Audrey|Audrey|Awesome",
            "Bergoglio, Jorge Mario|Jorge Mario|Bergoglio",
            "Cadili, Francesco|Francesco|Cadili",
            "Bergoglio|''|Bergoglio",
            "Francesco Cadili|Francesco|Cadili",
            "Philip Seymour Hoffman|Philip Seymour|Hoffman",
            "John Smith|John|Smith",
            "Guido van Rossum|Guido|van Rossum",
            "Francesco|Francesco|''",
            "Cadili|''|Cadili",
            "Jorge Mario Bergoglio|Jorge Mario|Bergoglio",
            "Jorge Bergoglio Mario Luigi|Jorge|Bergoglio Mario Luigi" // this works differently
    })
    @Test
    public void test(String fullname, String expectedFirstName, String expectedLastName) {
        assertThat(firstNames.extractFirstAndLastName(fullname))
            .returns(expectedFirstName, from(firstAndLastName -> firstAndLastName._1()))
            .returns(expectedLastName, from(firstAndLastName -> firstAndLastName._2()));
    }
```
And also it has a potential to be incorporated into our new solution for recognition if something is a Person or Organization